addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -790,14 +790,15 @@ export function getDeepRange(editable, { range, sel, splitText, select, correctT
     // at the last position of the previous node instead.
     const endLeaf = firstLeaf(end);
     const beforeEnd = endLeaf.previousSibling;
+    const previous = previousLeaf(endLeaf, editable, true);
     if (
         correctTripleClick &&
         !endOffset &&
         (start !== end || startOffset !== endOffset) &&
-        (!beforeEnd || (beforeEnd.nodeType === Node.TEXT_NODE && !isVisibleStr(beforeEnd)))
+        (!beforeEnd || (beforeEnd.nodeType === Node.TEXT_NODE && !isVisibleStr(beforeEnd))) &&
+        previous && !closestElement(previous, 'table')
```

Comment:
Why can't the leaf before the one at the end of the range be in a table?

---

Also, you might as well put the contents of the nested `if` in the top `if` then while you're at it.

```suggestion
        previous && !closestElement(previous, 'table') && closestElement(previous).isContentEditable)
    ) {
```